hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/clientrm/TestFederationClientInterceptor.java

```diff
@@ -641,4 +643,18 @@ public void testGetApplicationsApplicationStateNotExists() throws Exception{
     Assert.assertNotNull(responseGet);
     Assert.assertTrue(responseGet.getApplicationList().isEmpty());
   }
+
+  @Test
+  public void testGetClusterNodesRequest() throws Exception {
+    LOG.info("Test FederationClientInterceptor : Get Cluster Nodeds request");
+    // null request
+    LambdaTestUtils.intercept(YarnException.class,
+            "Missing getClusterNodes request.", () -> interceptor.getClusterNodes(null));
+
+    // normal request.
+    GetClusterNodesResponse response =
+            interceptor.getClusterNodes(GetClusterNodesRequest.newInstance());
+    Assert.assertEquals(subClusters.size(),
```

Comment:
Can we assert some metrics?

Reply:
I will refine it in unit tests.

Reply:
I added unit tests related to getClusterNodes metrics in TestRouterMetrics.java